modules/integration/tests-integration/tests-server/src/test/java/org/wso2/mb/integration/tests/server/mgt/MetricsTestCase.java

```diff
@@ -0,0 +1,458 @@
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "Mean",
+		                       MetricAttribute.MEAN, null));
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "Maximum",
+		                       MetricAttribute.MAX, null));
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "Standard Deviation",
+		                       MetricAttribute.STDDEV, null));
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "50th Percentile",
+		                       MetricAttribute.P50, null));
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "75th Percentile",
+		                       MetricAttribute.P75, null));
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "95th Percentile",
+		                       MetricAttribute.P95, null));
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "98th Percentile",
+		                       MetricAttribute.P98, null));
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "99th Percentile",
+		                       MetricAttribute.P99, null));
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "999th Percentile",
+		                       MetricAttribute.P999, null));
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "Mean Rate",
+		                       MetricAttribute.MEAN_RATE, null));
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "Last Minute Rate",
+		                       MetricAttribute.M1_RATE, null));
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "Last 5 Minutes Rate",
+		                       MetricAttribute.M5_RATE, null));
+		metrics.add(new Metric(MetricType.TIMER, "org.wso2.mb.database.write", "Last 15 Minutes Rate",
+		                       MetricAttribute.M15_RATE, null));
+
+		metricList.setMetric(metrics.toArray(new Metric[metrics.size()]));
+		metricData =
+				metricsViewClient.findLastMetrics(metricList, source, String.valueOf(fromTime));
+
+		// Stopping subscribers and publishers
+
+		consumerClient.stopClient();
+		publisherClient.stopClient();
+
+		// Evaluating metrics data
+		BigDecimal value;
+		for (int i = 0; i < metricData.getMetadata().getNames().length; i++) {
+			value = metricData.getData()[0][i];
+			Assert.assertTrue(null != value,
+			                  "Metric [" + metricData.getMetadata().getNames()[i] +
+			                  "] value was not reported.");
+		}
+
+
+
+	}
+
+	/**
+	 * This method will force metric manager to collect metrics by invoking report() method
+	 * using remote jmx
+	 * @throws IOException
+	 * @throws MalformedObjectNameException
+	 */
+	private void invokeJMXReportOperation() throws IOException, MalformedObjectNameException{
+		JMXServiceURL url =
+				new JMXServiceURL("service:jmx:rmi://localhost:11111/jndi/rmi://localhost:9999/jmxrmi");
```

Comment:
The JMS client used in our tests are also using a hardcoded "localhost" currently. I am not sure whether this URL can be grabbed from the test automation framework and etc.